linera-core/src/chain_worker/state/attempted_changes.rs

```diff
@@ -238,15 +238,13 @@ where
     /// Processes a confirmed block (aka a commit).
     pub(super) async fn process_confirmed_block(
         &mut self,
-        certificate: Certificate,
+        certificate: ConfirmedBlockCertificate,
         blobs: &[Blob],
         notify_when_messages_are_delivered: Option<oneshot::Sender<()>>,
     ) -> Result<(ChainInfoResponse, NetworkActions), WorkerError> {
-        let CertificateValue::ConfirmedBlock { executed_block, .. } = certificate.value() else {
-            panic!("Expecting a confirmation certificate");
-        };
-        let block = &executed_block.block;
-        let block_height = block.height;
+        let executed_block = certificate.executed_block();
+        let block = executed_block.block.clone();
```

Comment:
Why not take a reference?

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/2832/commits/e2f4558e190047550dd9fdaf60b7162d9e7a5bc6.